dataprofiler/data_readers/csv_data.py

```diff
@@ -110,7 +110,7 @@ def _determine_has_header(data_as_str):
         try:
             has_header = sniffer.has_header(data_as_str)
         except csv.Error:
-            has_header = 0
+            has_header = 1
```

Comment:
I'm concerned about this change has pandas does not do header detection. Pandas makes the assumption that the first row is the header.

Reply:
My thought process is this: if the sniffer fails to detect the header, we can either set header to 0 (fix from this PR) or None (currently existed). The advantage of setting to 0 is:
- most of the time, the files have headers, so the error rate is less than setting it to None
- setting to 0 will reduce 1 first line, but won't affect the results of dp, while setting to None will make the potential header combined with the value rows, so the detection results of dp may be affected. E.g., header is 'a b c', and values are '1 2 3\n 4 5 6\n...'

Reply:
one other way is to raise an error if the sniffer fails, and ask user to input the header.

Reply:
So just so I'm understanding, we aren't going to use sniffer any differently, we're just going to begin implicitly assuming a header is present?

Reply:
Why do we get the exceptions?

Reply:
@ChrisWallace2020 : yes, which is helpful from my first point above.
@lettergram : in case that the sniffer fails to detect the header, in which case we need to set header to either 0 or 1

Reply:
@AnhTruong I believe I found a potential solution. The problem itself is very interesting and understanding why it was occurring helped in assessing the solution, hence why I stressed understanding it further to potentially be able to address the problem at its root. Basically, the problem occurs bc the delimiter cannot be determined (in these cases due to single col which has no delimiters). For whatever reason, as oppose to saying no delimiter found (or giving that option), the program just errors. I actually found a case log on their lib which discusses this same issue.

I instead found a potential solution which avoids this error, and should only be executed when the csv.Error occurs as a backup.
```python
# ERRORS
sniffer = csv.Sniffer()
sniffer.has_header('header\n' + "\n".join(map(str, range(100))))
# NO ERRORS
sniffer = csv.Sniffer()
sniffer._guess_delimiter = lambda *args, **kwargs: (' ', 0) # ONLY EXECUTE THIS IF WE FAILED and got csv.Error
sniffer.has_header('header\n' + "\n".join(map(str, range(100))))
```

I think we should consider this as a potential solution, but not 100% confident it is the correct solution.

Reply:
still have some error with csv files of text without spaces, but I think that may be the best solution that we can have now